### PR TITLE
PLAT-1338: Add avatar details component

### PR DESCRIPTION
**Pull request by mihaelpalfi**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1338)

#### Description
As per JIRA ticket

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/components/zen-avatar-details/zen-avatar-details.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  @Prop() readonly color: string = '#00528C';
+
+  /** Icon background color  */
+  @Prop() readonly background: string = '#D5E9FA';
+
+  /** Padding */
+  @Prop() readonly padding: PaddingShorthand = 'lg';
+
+  /** Spacing between icon and username  */
+  @Prop() readonly spacing: Size = 'md';
```
Comment (chavlji) [inline src/components/zen-avatar-details/zen-avatar-details.tsx:29]: Is this specified by a designers? Imo space should be fixed and always the same.

File: src/components/zen-avatar-details/zen-avatar-details.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  @Prop() readonly userName: string = '';
+
+  /** User email  */
+  @Prop() readonly email: string = '';
+
+  /** Icon color  */
+  @Prop() readonly color: string = '#00528C';
+
+  /** Icon background color  */
+  @Prop() readonly background: string = '#D5E9FA';
```
Comment (chavlji) [inline src/components/zen-avatar-details/zen-avatar-details.tsx:23]: Rename to `icon-background`

File: src/components/zen-avatar-details/zen-avatar-details.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  @Element() host: HTMLZenAvatarDetailsElement;
+
+  /** User name  */
+  @Prop() readonly userName: string = '';
+
+  /** User email  */
+  @Prop() readonly email: string = '';
+
+  /** Icon color  */
+  @Prop() readonly color: string = '#00528C';
```
Comment (chavlji) [inline src/components/zen-avatar-details/zen-avatar-details.tsx:20]: Rename to `icon-color`

File: src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  it('should render avatar details correctly', async () => {
+    const page = await newSpecPage({
+      components: [ZenAvatarDetails, ZenAvatarIcon, ZenText],
+      html: `<zen-avatar-details />`,
+    });
+    page.root.users = user;
+    expect(page.root.shadowRoot.querySelector('zen-avatar-icon')).toBeTruthy();
+
+    const textElements = page.root.shadowRoot.querySelectorAll('zen-text');
+    expect(textElements.length).toEqual(2);
```
Comment (chavlji) [inline src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx:25]: Shouldn't this be part of zen-avatar-icon unit tests? Seems like duplication.

File: src/components/zen-avatar/test/zen-avatar.spec.tsx [hunk 5]
```diff
@@ -52,33 +53,43 @@ export const users = [
...
+      components: [ZenAvatar, ZenAvatarIcon, ZenTooltip],
       html: `<zen-avatar />`,
     });
     page.root.users = user;
     await page.waitForChanges();
-    const expectedAvatarIcons = 2; // 1 + 1 since tooltip includes one avatar icon as well
-    expect(page.root.shadowRoot.querySelectorAll('zen-avatar-icon').length).toEqual(expectedAvatarIcons);
+
+    const target = page.root.shadowRoot.querySelector('zen-avatar-icon');
+    expect(target.shadowRoot.lastChild.textContent).toEqual('MA');
```
Comment (chavlji) [inline src/components/zen-avatar/test/zen-avatar.spec.tsx:73]: This should be tested in `zen-avatar-icon` unit tests?

**Review by chavlji — CHANGES_REQUESTED** [review]
Looks good in general!

A few things to fix:
1. first letters are eaten-up from left side (screenshot)
2. if I set height to 300px, content should be vertically centered:
Screenshot 2021-03-11 at 12 42 44

File: src/components/zen-avatar-details/zen-avatar-details.tsx (hunk 1 above)
Comment (mihaelpalfi) [reply src/components/zen-avatar-details/zen-avatar-details.tsx:29]: I added it that we have more customisability. Designers can change this in a next design.

File: src/components/zen-avatar-details/zen-avatar-details.tsx (hunk 2 above)
Comment (mihaelpalfi) [reply src/components/zen-avatar-details/zen-avatar-details.tsx:23]: done

File: src/components/zen-avatar-details/zen-avatar-details.tsx (hunk 3 above)
Comment (mihaelpalfi) [reply src/components/zen-avatar-details/zen-avatar-details.tsx:20]: done

File: src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx (hunk 4 above)
Comment (mihaelpalfi) [reply src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx:25]: Zen avatar icon is just an icon that shows an image or initials.

File: src/components/zen-avatar/test/zen-avatar.spec.tsx (hunk 5 above)
Comment (mihaelpalfi) [reply src/components/zen-avatar/test/zen-avatar.spec.tsx:73]: Yes and it is tested there. So removed here.

**Comment by mihaelpalfi** [comment]
> Looks good in general!

> 1. first letters are eaten-up from left side (screenshot)
> 2. if I set height to 300px, content should be vertically centered:

1. Cant reproduce. Tested in chrome and firefox. Probably was an issue because of the space pr.
2. Why is this important? But if it is. I can change it like : 
`:host {
  display: block;
  display: flex;
  justify-content: center;
  align-items: center;
}`

File: src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx (hunk 4 above)
Comment (chavlji) [reply src/components/zen-avatar-details/test/zen-avatar-details.spec.tsx:25]: My bad... sorry.

File: src/components/zen-avatar-details/zen-avatar-details.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  @Prop() readonly iconColor: string = '#00528C';
+
+  /** Icon background color  */
+  @Prop() readonly iconBackground: string = '#D5E9FA';
+
+  /** Padding */
+  @Prop() readonly padding: PaddingShorthand = 'lg';
+
+  /** Spacing between icon and username  */
+  @Prop() readonly spacing: Size = 'md';
```
Comment (chavlji) [inline src/components/zen-avatar-details/zen-avatar-details.tsx:29]: I'd disagree to that. We shoul keep props on minimum and only make customizable stuff what are actually used in any design. Components should be flexible just enoguh to cover all use-cases and no more.

**Review by chavlji — APPROVED** [review]
Nice!

**Comment by chavlji** [comment]
1. Cant reproduce. Tested in chrome and firefox. Probably was an issue because of the space pr.
2. Why is this important? But if it is. I can change it like :
:host { display: block; display: flex; justify-content: center; align-items: center; }

1. Me neither... maybe it was zen-space problem 🤷🏻 
2. Please add it yeah. Because this is how you'd expect to behave if you'd change height of host?

**Comment by chavlji** [comment]
Please update e2e screenshot.

**Review by f-jauregui — APPROVED** [review]
Great work! 🥇

**Pull request merged**